Jetpack Scan: Don't show '<action> on <date>' if no date is available

mattgawarecki opened the pull request:
#### Changes proposed in this Pull Request

* Add a `Number.isFinite` check to `fixed_on` and `first_detected` API response properties, so that we don't create invalid dates for the UI to display.

#### Testing instructions

* In either Jetpack Cloud or Calypso, select a site that has Jetpack Scan and at least one fixed or ignored threat.
* Navigate to **Jetpack > Scan > History**.
* In React DevTools, check the `status` and `fixedOn` properties of any `ThreatItem` component.
  * Verify that if `fixedOn` is a numeric value, text is displayed to show when action was taken on the threat (see reference screenshots).
  * Verify that if `fixedOn` is undefined or non-numeric, no "<action taken> on <date>" text is displayed.
* Inspect different fixed/ignored threats to ensure behavior is correct in all cases (i.e., fixed and ignored threats, both with and without "fixed on" dates).
* **Alternatively:** In React DevTools, modify the values of a `ThreatItem`'s `status` and `fixedOn` properties to simulate different scenarios.

#### Reference screenshots

<img width="728" alt="Screen Shot 2020-10-07 at 08 40 06" src="https://user-images.githubusercontent.com/670067/95338799-eb32ba00-0878-11eb-9b4c-472e8ecfc955.png">
<img width="728" alt="Screen Shot 2020-10-07 at 08 41 58" src="https://user-images.githubusercontent.com/670067/95338845-fb4a9980-0878-11eb-83cd-7b98e5d9b4fb.png">

<img width="727" alt="Screen Shot 2020-10-07 at 08 43 15" src="https://user-images.githubusercontent.com/670067/95339005-2af9a180-0879-11eb-8ca8-c9f5e3fba6d9.png">
<img width="729" alt="Screen Shot 2020-10-07 at 08 43 05" src="https://user-images.githubusercontent.com/670067/95339011-2d5bfb80-0879-11eb-941b-8acd4f92f1ec.png">

blackjackkent commented:
Would it be good to include some sort of text for this scenario but not including a date? (At a quick glance down the example screenshot you provided, my immediate visual interpretation was that the second item was not in the same state as the first since it didn't have that message, even though on a second look I could see they both have the "Fixed" label.)

Maybe just "Threat fixed" or "Resolved" or something like that?

monsieur-z reviewed (approved):
LGTM! I tested the 4 different combinations of statuses and dates.

mattgawarecki commented:
@blackjackkent Addressing your feedback :-) @monsieur-z Can you verify these messages show up without a date when `threat.fixedOn` isn't defined?

<img width="697" alt="image" src="https://user-images.githubusercontent.com/670067/95618807-46a8a780-0a33-11eb-9138-9c54d5ecaa7f.png">

<img width="696" alt="image" src="https://user-images.githubusercontent.com/670067/95618911-63dd7600-0a33-11eb-90ac-baa3035bfc00.png">

a8ci18n commented on client/components/jetpack/threat-item-subheader/index.tsx:
```diff
@@ -33,9 +33,41 @@ const formatDate = ( date: Date ) => {
+					args: { date },
+					comment: 'Past tense action: a threat was ignored on a specific date',
+			  } )
+			: translate( 'Threat ignored', {
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'Threat ignored.' )` *ES Score: 13*

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/4968162

Thank you @mattgawarecki for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/4968162).
